lib/useragent.ts

```diff
@@ -1,13 +1,217 @@
-const useragent = require('express-useragent-x');
+const useragent = require('express-useragent');
 
-export default class UserAgent {
-  private readonly agent: any
+export default class UserAgent implements ExpressUseragent.UserAgent {
+    get isEpiphany() {
+        return this.agent.isEpiphany;
+    }
 
-  constructor(source = 'unknown') {
-    this.agent = useragent.parse(source);
+    get isDesktop(): boolean {
+        return this.agent.isDesktop;
+    }
 
-    return this.agent;
-  }
-}
+    get isWindows(): boolean {
+        return this.agent.isWindows;
+    }
+
+    get isWindowsPhone(): boolean {
+        return this.agent.isWindowsPhone;
+    }
+
+    get isLinux(): boolean {
+        return this.agent.isLinux;
+    }
+
+    get isLinux64(): boolean {
+        return this.agent.isLinux64;
+    }
+
+    get isMac(): boolean {
+        return this.agent.isMac;
+    }
+
+    get isChromeOS(): boolean {
+        return this.agent.isChromeOS;
+    }
+
+    get isBada(): boolean {
+        return this.agent.isBada;
+    }
+
+    get isSamsung(): boolean {
+        return this.agent.isSamsung;
+    }
+
+    get isRaspberry(): boolean {
+        return this.agent.isRaspberry;
+    }
+
+    // @types/express-useragent's type definition is not so correct
+    get isBot(): any {
+        return this.agent.isBot;
+    }
+
+    get isCurl(): boolean {
+        return this.agent.isCurl;
+    }
+
+    get isAndroidTablet(): boolean {
+        return this.agent.isAndroidTablet;
+    }
+
+    get isWinJs(): boolean {
+        return this.agent.isWinJs;
+    }
+
+    get isKindleFire(): boolean {
+        return this.agent.isKindleFire;
+    }
+
+    get isSilk(): boolean {
+        return this.agent.isSilk;
+    }
+
+    get isCaptive(): boolean {
+        return this.agent.isCaptive;
+    }
+
+    get isSmartTV(): boolean {
+        return this.agent.isSmartTV;
+    }
+
+    get silkAccelerated(): boolean {
+        return this.agent.silkAccelerated;
+    }
+
+    get SilkAccelerated(): boolean {
+        return this.agent.silkAccelerated;
+    }
+
+    get browser(): string {
+        return this.agent.browser;
+    }
+
+    // @types/express-useragent's type definition is not so correct
+    get version(): any {
+        return this.agent.version;
+    }
+
+    get os(): string {
+        return this.agent.os;
+    }
+
+    get platform(): string {
+        return this.agent.platform;
+    }
+
+    get geoIp(): any {
+        return this.agent.geoIp;
+    }
+
+    get source(): string {
+        return this.agent.source;
+    }
+
+    private readonly agent: any
```

Comment:
It's wrong because useragent.parse() have return type, also better to use undescore prefix for private members because after TS compile you have access to it in js

Reply:
I've added some comment on this line to explain that will use any for now because @types/express-useragent has wrong definitions on this.